### DISCUSSION TITLE
Please use different colors to mark function parameters and function annotations so that I can quickly browse function signatures

### DISCUSSION THREAD

**Discussion by CatNofishing** [microsoft/pylance-release#2685] [Enhancement]
Function parameters and function annotations are marked in the same color, so I can't quickly distinguish between default parameters and function annotations
![image](https://user-images.githubusercontent.com/33288114/99754442-5aabf280-2b23-11eb-9f92-c7e16f44cdb0.png)

**Comment by huguesv** [comment]
Thank you for the suggestion.

This is a case where the textmate grammar is getting confused because we send Python markdown code section but it's not really valid Python.

![image](https://user-images.githubusercontent.com/1696845/99837154-f5a0ce80-2b1b-11eb-8bfb-d7886fa8df7a.png)

If we change the `(class)` with `def` it becomes more valid and we can see the parameters get colorized correctly:

![image](https://user-images.githubusercontent.com/1696845/99837219-1537f700-2b1c-11eb-9a00-1e77b10e9c74.png)

**Comment by CatNofishing** [comment]
Any plan to improve this issue?😁

**Comment by jakebailey** [comment]
We are working on the tooltips, yes.

VS Code provides no mechanism to customize the colors of these tooltips. They are markdown that have no access to semantic tokens, just the TextMate profile.

I've tried changing our tooltips to a custom grammar that can handle the `(function)` or `(class)` etc at the beginning, but VS Code hardcodes the list of valid languages in tooltips, making it impossible.

This may be better if we drop `(class)` or something, as noted above.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
